source/extensions/filters/http/cache/cache_headers_utils.h

```diff
@@ -114,10 +114,6 @@ class CacheHeadersUtils {
 
 class VaryHeader {
 public:
-  // Checks if the headers contain an allowed value in the Vary header.
-  static bool isAllowed(const std::vector<Matchers::StringMatcherPtr>& allowlist,
-                        const Http::ResponseHeaderMap& headers);
-
   // Checks if the headers contain a non-empty value in the Vary header.
   static bool hasVary(const Http::ResponseHeaderMap& headers);
```

Comment:
Should these static methods still be public? Or will they now be used only internal to the class as helpers?

Reply:
`hasVary` and `createVaryKey` are used by `simple_http_cache`, so those should stay public. I'll make the other two methods private.

Reply:
I moved the logic of `parseAllowlist` to the constructor and made `parseHeaderValues` more generic and moved it into `CacheHeaderUtils`.
Let me know if that's okay.